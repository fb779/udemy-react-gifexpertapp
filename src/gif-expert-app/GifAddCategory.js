import React, {useState} from 'react';
import PropTypes from 'prop-types';

const GifAddCategory = ({setcategories}) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (!inputValue) return;
    setcategories((catg) => {
      if (catg.includes(inputValue)) return catg;
      return [inputValue, ...catg];
    });
    setInputValue('');
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} placeholder="Valor por defecto" />
    </form>
  );
};

GifAddCategory.propTypes = {
  setcategories: PropTypes.func.isRequired,
};

export default GifAddCategory;
