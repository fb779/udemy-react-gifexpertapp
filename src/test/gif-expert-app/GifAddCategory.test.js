import React from 'react';
import '@testing-library/jest-dom';

import {shallow} from 'enzyme';
import GifAddCategory from '../../gif-expert-app/GifAddCategory';

describe('prubeas en <GifAddCategory />', () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<GifAddCategory setcategories={setCategories} />);

  beforeEach(() => {
    wrapper = shallow(<GifAddCategory setcategories={setCategories} />);
  });

  test('Carga del componenete <GifAddCategory /> ', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe cambiar el valor de la caja de texto', () => {
    const input = wrapper.find('input');
    const value = 'saitama';

    input.simulate('change', {target: {value}});

    expect(wrapper.find('p').text().trim()).toBe(value);
  });

  test('No debe de postear la informaciÃ³n con submit ', () => {
    wrapper.find('form').simulate('submit', {preventDefault() {}});

    expect(setCategories).not.toHaveBeenCalled();
  });

  test('debe de llamar al setCategory y limpiar la caja de texto ', () => {
    const value = 'saitama';

    wrapper.find('input').simulate('change', {target: {value}});
    wrapper.find('form').simulate('submit', {preventDefault() {}});

    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

    expect(wrapper.find('input').prop('value').trim()).toBe('');
  });
});
