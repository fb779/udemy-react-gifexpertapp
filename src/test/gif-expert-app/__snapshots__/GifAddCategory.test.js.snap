// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prubeas en <GifAddCategory /> Carga del componenete <GifAddCategory />  1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Valor por defecto"
    type="text"
    value=""
  />
</form>
`;
